main.go

```diff
@@ -104,31 +133,51 @@ func main() {
 		log.Fatalf("Failed to get instance of the helmrelease with %v", err)
 	} else if err != nil {
 		// This means that the object was not found
-		log.Infof("Did not find the helm release: %v, creating the release...", hr.Name)
-		// Create the HelmRelease if it doesn't exist
-		retry(ctx, func() error { return clientSet.Create(ctx, hr) }, interval)
+		log.Infof("Did not find the helm release: %v", hr.Name)
+		if action == Install {
+			// Create the HelmRelease if it doesn't exist
+			log.Infof("Creating the release: %v...", hr.Name)
+			retry(ctx, func() error { return clientSet.Create(ctx, hr) }, interval)
+		} else {
```

Comment:
I don't think we need this `else` considering we are already validating the params above